### ISSUE TITLE
Library pings hostname instead of IP

### ISSUE THREAD

**Issue by SemenovAlexander** [stealthcopter/AndroidNetworkTools#8]
Because of calling `getHostName()` instead of `getHostAddress()` [on this line](https://github.com/stealthcopter/AndroidNetworkTools/blob/master/library/src/main/java/com/stealthcopter/networktools/ping/PingNative.java#L23), ping will always try to use hostname instead of ip, even if user intended to ping IP directly. 

It's not an issue if you parse ip from String input, however, if you try to perform ping of `InetAddress` object, obtained from NSLookup (it will contain hostname), actual address will depend on DNS in case of native ping.
And ping result will report actually requested ip, because it's conutructed [before](https://github.com/stealthcopter/AndroidNetworkTools/blob/master/library/src/main/java/com/stealthcopter/networktools/ping/PingNative.java#L16) actual ping.

I'm not sure if it's completely fine to just replace `getHostName()` with `getHostAddress()`, because it can lead to different results as native ping, if user pings hostname.

As for java ping, this seems to be not an issue.

**Comment by stealthcopter** [comment]
Ok, good find. I suggest that I change the PingNative code to something like below to catch for this and to also add support for ipv6 addresses as the same time.

```
String address = host.getHostAddress();
String pingCommand = "ping";

if (address != null) {
    if (IPTools.isIPv6Address(address)) {
        // If we detect this is a ipv6 address, change the to the ping6 binary
        pingCommand = "ping6";
    } else if (!IPTools.isIPv4Address(address)) {
        // Address doesn't look to be ipv4 or ipv6, but we could be mistaken
        Log.w("AndroidNetworkTools", "Could not identify " + address + " as ipv4 or ipv6, assuming ipv4");
    }
} else {
    // Not sure if getHostAddress ever returns null, but if it does, use the hostname as a fallback
    address = host.getHostName();
}

Process proc = runtime.exec(pingCommand + " -c 1 -w " + timeoutSeconds + " " + address);

```

Let me know if you have any thoughts on the code, otherwise I'll included in into the library shortly.

Thanks :)

**Comment by SemenovAlexander** [comment]
Yeap, code looks great.

I think, it would be nice to let user some config option in `Ping` object builder to use hostname vs hostaddress during native ping. 

 Glad I could help :)

**Comment by stealthcopter** [comment]
This is now fixed in the latest release 0.2.0
